### PR TITLE
docs: add GITPOD.md. overhaul contributing dev workflow.

### PR DESCRIPTION
**Pull request by Samyak2**
documented Gitpod and dev docker compose in contributing.

GITPOD.md contains all details on gitpod.

The PR is targeted to main since it's a documentation change.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,14 +53,41 @@ Follow these steps if you want to make a code contribution to Chaos Genius.
 
 ## Development Workflow
 
-### Backend/API
+We support 3 different development workflows:
+- [Gitpod](#gitpod): easiest and fastest way to get started. Has some minor limitations but can be used in most cases.
```
Comment (rjdp) [inline CONTRIBUTING.md:57]:
```suggestion
- [Gitpod](#gitpod): easiest and fastest way to get started. Has some minor [limitations](GITPOD.md#known-issues) but can be used in most cases.
```

**Review by rjdp — CHANGES_REQUESTED** [review]
LGTM!

File: CONTRIBUTING.md (hunk 1 above)
Comment (Samyak2) [reply CONTRIBUTING.md:57]: Good idea. Thanks

**Review by rjdp — APPROVED** [review]
LGTM !

**Pull request merged**